q_learning.py

```diff
@@ -67,26 +72,33 @@ def runExperiment(env, num_episodes, agent, render):
             (sp, r, done, __) = env.step(a)  # Note: the environment "registers" the new sp as env.pos
             agent.update(s, sp, r, a, done)
 
-            # For debugging: store the t-distribution variance and action_values at each timestep:
-            # s_idx = np.ravel_multi_index(s, agent.state_shape)
-            # x = agent.getIndex(s) + (a * num_states)
-            # nu = agent.bayesianQ.B[x, 1]
-            # alpha = agent.bayesianQ.B[x, 2]
-            # beta = agent.bayesianQ.B[x, 3]
-            # t_distribution_var = beta * (nu + 1) / (alpha * nu)
-            # t_dist_vars[a][s_idx].append(t_distribution_var)
+            # For debugging: store the t-distribution variance at each timestep:
+            # mu, nu, alpha, beta = self.B[x, :]
+            s_idx = np.ravel_multi_index(s, agent.state_shape)
+            x = agent.getIndex(s) + (a * num_states)
+            nu = agent.rewardApprox.B[x, 1]
+            alpha = agent.rewardApprox.B[x, 2]
+            beta = agent.rewardApprox.B[x, 3]
+            t_distribution_var = beta * (nu + 1) / (alpha * nu)
+            t_dist_vars[a][s_idx].append(t_distribution_var)
+            bonuses[a][s_idx] = agent.rewardApprox.bonus
             # ac_vals_list.append(agent.act_vals)
 
             # For debugging: store the maximum variance for each timestep:
-            # var_single_timestep = []
-            # for x_i in range(num_states * num_acts):
-            #     nu_i = agent.bayesianQ.B[x_i, 1]
-            #     alpha_i = agent.bayesianQ.B[x_i, 2]
-            #     beta_i = agent.bayesianQ.B[x_i, 3]
-            #     t_distribution_var = beta_i * (nu_i + 1) / (nu_i * alpha_i)
-            #     var_single_timestep.append(t_distribution_var)
-            # max_var_T.append(np.max(var_single_timestep))
-            # data_dict = {'max_var_T_array': max_var_T}
+            var_single_timestep = []
+            bonus_single_timestep = []
+            for x_i in range(num_states * num_acts):
+                nu_i = agent.rewardApprox.B[x_i, 1]
+                alpha_i = agent.rewardApprox.B[x_i, 2]
+                beta_i = agent.rewardApprox.B[x_i, 3]
+                t_distribution_var = beta_i * (nu_i + 1) / (nu_i * alpha_i)
+                var_single_timestep.append(t_distribution_var)
+                bonus_single_timestep.append(agent.rewardApprox.bonus)
```

Comment:
Which variable is this? An instance variable of the TabularBayesianApproximator? I don't see where it's initialized though...